core/common/utils/StringUtilities.cpp

```diff
@@ -135,4 +135,26 @@ namespace netmeld::core::utils {
 
     return ifaceName;
   }
+
+  std::string
+  compileCPE(const std::string& _part, const std::string& _vendor,
+             const std::string& _product, const std::string& _version)
+  {
+    // CPE format follows NMAP standard: 
+    // cpe:/{part}:{vendor}:{product}:{version}
+
+    std::string cpe {"cpe:/"};
+    cpe += _part + ":" + _vendor + ":" + _product + ":" + _version;
+    return cpe;
```

Comment:
FWIW (and not a required change), this seems like it would be much clearer if you used `std::format()` instead.  So something like.
```suggestion
    return std::format("cpe:/{}:{}:{}:{}", _part, _vendor, _product, _version);
```